src/util/account_util.js

```diff
@@ -212,18 +212,30 @@ async function generate_account_keys(req) {
     if (account.is_support) {
         throw new RpcError('FORBIDDEN', 'Cannot update support account');
     }
-    const access_keys = cloud_utils.generate_access_keys();
-    const decrypted_access_keys = _.cloneDeep(access_keys);
-    access_keys.secret_key = system_store.master_key_manager.encrypt_sensitive_string_with_master_key_id(
-        access_keys.secret_key, account.master_key_id._id);
-
+    const account_access_keys = account.access_keys || [];
+    // Encrypt the secret_key before saving it againg other wise secret_key saved without encrypt
+    if (account_access_keys.length > 0) {
```

Comment:
I didn't understand what is the condition and what we try to do here.
Could you please explain here and also add comments so it would be clear?